libs/vec/native/src/vec/c/vec.c

```diff
@@ -60,12 +53,26 @@ EXPORT int32_t dot8s(int8_t* a, int8_t* b, size_t dims) {
     return vaddvq_s32(vaddq_s32(acc5, acc6));
 }
 
-EXPORT int32_t sqr8s(int8_t *a, int8_t *b, size_t dims) {
+EXPORT int32_t dot8s(int8_t* a, int8_t* b, size_t dims) {
+    int32_t res = 0;
+    int i = 0;
+    if (dims > DOT8_STRIDE_BYTES_LEN) {
+        i += dims & ~(DOT8_STRIDE_BYTES_LEN - 1);
```

Comment:
This only works if DOT8_STRIDE_BYTES_LEN is a power of 2. Of course it will be. Perhaps it is worth enforcing this with a static_assert somewhere so people don't accidentally break it though, i.e. `static_assert((1031 & ~(DOT8_STRIDE_BYTES_LEN - 1)) == (1031 - 1031 % DOT8_STRIDE_BYTES_LEN), "Invalid DOT8_STRIDE_BYTES_LEN must be a power of 2");`. Note this can be anywhere in the source file, so you don't need to put in any actual function definition (although it shouldn't actually generate any code).

Reply:
++
I think you can even make this a compile time error (not sure if you need to switch to c++ for that though)